### PR TITLE
added form for database insertion

### PR DESCRIPTION
**Pull request by touredri**
- [ ] Created a method called current_user in your ApplicationController
- [ ] Created forms to perform the following functions:
  - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
  - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- [ ] Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   root 'users#index'
+
   resources :users, only: %i[index show] do
     resources :posts, only: %i[index show]
   end
+  resources :posts, only: %i[new create] do
+    resources :comments, only: %i[new create]
+  end
```
Comment (codepantha) [inline config/routes.rb:12]: - [ ] Add ability to like a post. For this to happen, there needs to be a `LikesController` with a `create` method that adds a like to a given post. You also need to set up the correct routing which should handle a POST request `/posts/:post_id/likes`

**Comment by brytebee** [comment]
@touredri Is this the first time you're requesting a review for this milestone? It appears as a re-review.

**Comment by touredri** [comment]
the second @brytebee , but I don't know why the first reviewer didn't mention anything here

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  def current_user
+    # will be available in all controllers
+    @current_user = User.find(1)
```
Comment (brytebee) [inline app/controllers/application_controller.rb]: Kindly update this line t `User.first`. Models can be filtered by adding such ruby methods such as first, second, last, etc.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @touredri 👋,

##

### Highlights💡
- Great work connecting your app to a DB.✔️
- Great understanding of the flow of data from controller action to views.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (touredri) [reply app/controllers/application_controller.rb]: @brytebee ok

Comment (Whoistolu) [reply app/controllers/application_controller.rb]: - [ ] [REQUIRED]n Hello @touredri , as suggested by the previous reviewer, let's update the code for the `current_user` method to: 

```ruby
def current_user
  User.first
end
```

### Here is a screenshot from the project requirement
Screenshot 2023-07-22 at 16 29 59

### It is required to return the first user from the database and not find the user with ID of 1.

The difference is that your code ` @current_user = User.find(1)` specifically finds a User record with ID 1 and stores it in the @current_user variable. In contrast, the code we recommend only retrieves the first User record from the database but does not store it for future reference.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @touredri 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (touredri) [reply app/controllers/application_controller.rb]: oh I done but not added to commit

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @touredri 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- work professionally presented
- great job connecting to db
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**